src/scippnexus/nxobject.py

```diff
@@ -320,9 +305,18 @@ def _get_child(
             da.coords['depends_on'] = t if isinstance(t, sc.Variable) else sc.scalar(t)
         return da
 
+    def _get_children_by_nx_class(self, select: type) -> Dict['NXobject']:
+        children = {}
+        for key in self.keys():
+            if (child := self._get_child(key)).attrs.get('NX_class') == select.__name__:
+                children[key] = child
+        return children
+
     def __getitem__(
             self,
             name: NXobjectIndex) -> Union['NXobject', Field, sc.DataArray, sc.Dataset]:
```

Comment:
```suggestion
            name: Union[NXobjectIndex, type]) -> Union['NXobject', Field, sc.DataArray, sc.Dataset, Dict['Nxobject']]:
```
Or cleaner, use https://docs.python.org/3/library/typing.html#typing.overload